### PR TITLE
docs: add deprecation warning

### PR DESCRIPTION
**Pull request by samouss**
**Summary**

Add a deprecation banner for the current documentation with a link on the next version documentation website.

**Result**

screen shot 2018-01-31 at 09 28 36

### CODE REVIEW

**Comment by samouss** [comment]
I think we should link directly on the migration guide instead of the homepage. WDYT?

**Comment by Haroenv** [comment]
The migration guide could be very prominent on the site, or in another banner?

**Comment by samouss** [comment]
What do you mean? On which website?

File: docgen/layouts/common/header.pug [hunk 1]
```diff
@@ -1 +1,6 @@
 div !{header}
+  .marketing-banner
+    .banner__description
+      a(href='http://react-instantsearch-beta.netlify.com')
+        p
+          strong Checkout React InstantSearch V5! Our new and improved React InstantSearch version →
```
Comment (Haroenv) [inline docgen/layouts/common/header.pug]: Check out

**Comment by samouss** [comment]
@Haroenv @vvo What about adding a yellow banner like the one on [InstantSearch.js]() for the v4 website. And for the v5 website we use the blue banner for notice the user that he is on the next version of the documentation. We also add a link to the migration guide inside it. WDYT?

**Comment by Haroenv** [comment]
still necessary to add the banner to the beta site

**Pull request merged**